### PR TITLE
Fix #10287 - ExpandObjects - Incorrect warning about Boiler:HotWater having too many fields

### LINKED ISSUE #10287: ExpandObjects - Incorrect warning about Boiler:HotWater having too many fields
Issue overview
--------------

An extra field was added for Boiler:HotWater in #9925

```shell
$ /Applications/EnergyPlus-23-2-0/energyplus -D -d out -x /Applications/EnergyPlus-23-2-0/ExampleFiles/VAVSingleDuctConstFlowBoiler.idf

$ cat out/eplusout.experr
ExpandObjects: SEVERE: Too many fields were found for the object: Boiler:HotWater, Number expected=[17], Number found=[18]
```
### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): all
 - Version of EnergyPlus (if using an intermediate build, include SHA): 23.2.0

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

* Fix #10287

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/ExpandObjects/epfilter.f90 [hunk 1]
```diff
@@ -1507,7 +1507,7 @@ PROGRAM epfilter
 INTEGER,PARAMETER :: blrhwNameOff = 1
 INTEGER,PARAMETER :: blrhwInletNodeOff = 11
 INTEGER,PARAMETER :: blrhwOutletNodeOff = 12
-INTEGER,PARAMETER :: blrhwLastFieldOff = 17
+INTEGER,PARAMETER :: blrhwLastFieldOff = 18
```
Comment (jmarrec) [inline src/ExpandObjects/epfilter.f90:1510]: Extra field was added, total number is 18

File: src/ExpandObjects/epfilter.f90 [hunk 2]
```diff
@@ -2305,7 +2305,7 @@ SUBROUTINE SetUpObjectsToProcess(doGatherSurfaces)
 CALL AddObjToProcess('HVACTemplate:Plant:MixedWaterLoop',.TRUE.,    mwpNameOff,                  mwpLdDistSchmOff,          17)
 CALL AddObjToProcess('HVACTemplate:System:DedicatedOutdoorAir',.TRUE.,doasNameOff,              doasHumidSetPtSchedNameOff, 46)
 ! Detailed Plant Objects
-CALL AddObjToProcess('Boiler:HotWater',.FALSE.,                     blrhwOutletNodeOff,          blrhwLastFieldOff,         17)
+CALL AddObjToProcess('Boiler:HotWater',.FALSE.,                     blrhwOutletNodeOff,          blrhwLastFieldOff,         18)
```
Comment (jmarrec) [inline src/ExpandObjects/epfilter.f90:2308]: I'm not sure whether I need to bump 17 to 18 here or not.

**Comment by jmarrec** [comment]
Also not sure if I need to do `    CALL AddToObjStr('Off Cycle Parasitic Fuel Load {W}','0')` after https://github.com/NREL/EnergyPlus/blob/d07d7ad64207285b241ab79b67b66ed98e2db594/src/ExpandObjects/epfilter.f90#L26185

The End Use Subcategory isn't being explicitly written. But the Off Cycle does not have a default (though it defaults to 0.0 W in the C++ code... I wish the E+.idd did have an explicit `\default 0.0`

**Comment by jmarrec** [comment]
Before:

```shell
$ $ep_build/Products/energyplus -x -D -d out-ori VAVSingleDuctConstFlowBoiler.idf
ExpandObjects Started.
No expanded file generated.
ExpandObjects Finished with Error(s). Time:     0.023
EnergyPlus Starting

[...]

$ cat out-ori/eplusout.experr 
ExpandObjects: SEVERE: Too many fields were found for the object: Boiler:HotWater, Number expected=[17], Number found=[18]
```

After

```shell
$ $ep_build/Products/energyplus -x -D -d out-new VAVSingleDuctConstFlowBoiler.idf
ExpandObjects Started.
No expanded file generated.
ExpandObjects Finished. Time:     0.014
EnergyPlus Starting

[...]

$ cat out-new/eplusout.experr
cat: out-new/eplusout.experr: No such file or directory
```

**Review by mjwitte — APPROVED** [review]
Looks good. Thanks @jmarrec

**Pull request merged**